File: benchmark/hashmap/hashmap.py
N = 300000
KEYS = 20011
REPS = 25

size_sum = 0
total = 0
for rep in range(REPS):
    counts = {}
    for i in range(N):
        key = 'k' + str((i % KEYS) * 7919 % KEYS)
        if key in counts:
            counts[key] += 1
        else:
            counts[key] = 1

    for i in range(N):
        key = 'k' + str((i % KEYS) * 7919 % KEYS)
        total += counts[key]

    size_sum += len(counts)

print(size_sum)
print(total)

File: benchmark/hashmap/hashmap.pl
use strict;
use warnings;

my $n = 300000;
my $keys = 20011;
my $reps = 25;

my $size_sum = 0;
my $total = 0;
for my $rep (1 .. $reps) {
    my %counts;
    for my $i (0 .. $n - 1) {
        my $key = 'k' . (($i % $keys) * 7919 % $keys);
        if (exists $counts{$key}) {
            $counts{$key} += 1;
        } else {
            $counts{$key} = 1;
        }
    }

    for my $i (0 .. $n - 1) {
        my $key = 'k' . (($i % $keys) * 7919 % $keys);
        $total += $counts{$key};
    }

    $size_sum += scalar keys %counts;
}

print "$size_sum\n";
print "$total\n";
